### ISSUE TITLE
update error return type on keeper Authority check to govtypes.InvalidSigner error

### ISSUE THREAD

**Issue by charleenfei** [cosmos/ibc-go#4815]
it looks like theres an invalid signer error in `govtypes`. maybe we can update all of these to use that error return, looks like that's what the 08-wasm client returns in `StoreCode`. i can open a code hygiene issue.

_Originally posted by @charleenfei in https://github.com/cosmos/ibc-go/pull/4773#discussion_r1345666125_

**Comment by damiannolan** [comment]
I think we had decided previously to not rely on the `govtypes` error and instead use our own. Just purely due to it being an error maintained outside of our module. 

I remember agreeing on this format somewhere in a PR thread or something: `errorsmod.Wrapf(ibcerrors.ErrUnauthorized, "expected %s, got %s", k.GetAuthority(), msg.Signer)`

**Comment by crodriguezvega** [comment]
I agree with @damiannolan. I also remember this being discussed and agreed in some PR.

I have [updated 08-wasm also to align with the usage of `ibcerrors.ErrUnauthorized`](https://github.com/cosmos/ibc-go/commit/8c9c7174be65a2c8b8962c20a03d597f2584a200), so I think we can close the issue.

**Comment by charleenfei** [comment]
got it, i don't know if i was in that discussion or maybe just forgot. thanks!
